wallet-new/src/Cardano/Wallet/Kernel/DB/Sqlite.hs

```diff
@@ -0,0 +1,562 @@
+
+instance Database Sqlite MetaDB
+
+{--
+
+* Table 1: ~tx_metas~
+** Primary Index: ~tx_meta_id~
+** Secondary Indexes (for now): ~tx_meta_created_at~
+
+| tx_meta_id | tx_meta_amount | tx_meta_created_at | tx_meta_is_local | tx_meta_is_outgoing |
+|------------+----------------+--------------------+------------------+---------------------|
+| Core.TxId  | Core.Coin      | Core.Timestamp     | Bool             | Bool                |
+
+--}
+data TxMetaT f = TxMeta {
+      _txMetaTableId         :: Beam.Columnar f Core.TxId
+    , _txMetaTableAmount     :: Beam.Columnar f Core.Coin
+    , _txMetaTableCreatedAt  :: Beam.Columnar f Core.Timestamp
+    , _txMetaTableIsLocal    :: Beam.Columnar f Bool
+    , _txMetaTableIsOutgoing :: Beam.Columnar f Bool
+    } deriving Generic
+
+type TxMeta = TxMetaT Identity
+
+deriving instance Eq TxMeta
+deriving instance Show TxMeta
+
+-- | Creates a storage-specific 'TxMeta' out of a 'Kernel.TxMeta'.
+mkTxMeta :: Kernel.TxMeta -> TxMeta
+mkTxMeta txMeta = TxMeta {
+                  _txMetaTableId         = txMeta ^. Kernel.txMetaId
+                , _txMetaTableAmount     = txMeta ^. Kernel.txMetaAmount
+                , _txMetaTableCreatedAt  = txMeta ^. Kernel.txMetaCreationAt
+                , _txMetaTableIsLocal    = txMeta ^. Kernel.txMetaIsLocal
+                , _txMetaTableIsOutgoing = txMeta ^. Kernel.txMetaIsOutgoing
+                }
+
+instance Beamable TxMetaT
+
+instance Table TxMetaT where
+    data PrimaryKey TxMetaT f = TxIdPrimKey (Beam.Columnar f Core.TxId) deriving Generic
+    primaryKey = TxIdPrimKey . _txMetaTableId
+
+instance Beamable (PrimaryKey TxMetaT)
+
+{--
+
+* Table 2: ~tx_meta_inputs~
+** Primary Index: ~tx_meta_input_address~
+** Secondary Indexes: ~tx_meta_id~
+
+| tx_meta_input_address | tx_meta_coin | tx_meta_id |
+|-----------------------+--------------+------------|
+| Core.Address          | Core.Coin    | Core.TxId  |
+
+** Table 3: ~tx_meta_outputs~
+** Primary Index: ~tx_meta_output_address~
+** Secondary Indexes: ~tx_meta_id~
+
+| tx_meta_output_address | tx_meta_coin | tx_meta_id |
+|------------------------+--------------+------------|
+| Core.Address           | Core.Coin    | Core.TxId  |
+
+--}
+
+data TxCoinDistributionTableT f = TxCoinDistributionTable {
+      _txCoinDistributionTableAddress :: Beam.Columnar f Core.Address
+    , _txCoinDistributionTableCoin    :: Beam.Columnar f Core.Coin
+    , _txCoinDistributionTxId         :: Beam.PrimaryKey TxMetaT f
+    } deriving Generic
+
+type TxCoinDistributionTable = TxCoinDistributionTableT Identity
+
+instance Prelude.Show TxCoinDistributionTable where
+    show txCoin =
+      let (TxIdPrimKey txid) = _txCoinDistributionTxId txCoin
+      in formatToString ("id = " % build % ", address = " % build % ", coin = " % build)
+                        txid
+                        (_txCoinDistributionTableAddress txCoin)
+                        (_txCoinDistributionTableCoin txCoin)
```

Comment:
The reason why I went with a custom `Show` instance (which indeed I try to stir away from as well) was because deriving an instance of `TxPrimKey ...` was bringing me into a needless rabbit hole. Considering I don't think we are using this anyway (in fact, it is a relic of me trying to debug something else via `traceShow`) I'm inclined to simply remove it.